src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -382,7 +384,16 @@ object RuleUtils {
             baseRelation.schema.contains(s) || (filesDeleted.nonEmpty && s.name.equals(
               IndexConstants.DATA_FILE_NAME_ID))))
 
-        val newLocation = new InMemoryFileIndex(spark, filesToRead, Map(), None)
+        val newLocation = if (filesToRead.length == index.content.files.size) {
+          index.withCachedTag(IndexLogEntryTags.INMEMORYFILEINDEX_INDEX_ONLY) {
+            new InMemoryFileIndex(spark, index.content.files, Map(), None)
```

Comment:
Can `index.content.files` be replaced with `filesToRead`? In that case, we can just change tag based on the condition (easier to follow).